packages/lodestar/src/chain/beaconProposerCache.ts

```diff
@@ -0,0 +1,37 @@
+import {MapDef} from "../util/map";
+import {Epoch} from "@chainsafe/lodestar-types";
+import {IMetrics} from "../metrics";
+
+const PROPOSER_PRESERVE_EPOCHS = 2;
+export type ProposerPreparationData = {
+  validatorIndex: string;
+  feeRecipient: string;
+};
+
+export class BeaconProposerCache {
+  private readonly feeRecipientByValidatorIndex: MapDef<string, {epoch: Epoch; feeRecipient: string}>;
+  constructor(opts: {defaultFeeRecipient: string}, private readonly metrics?: IMetrics | null) {
+    this.feeRecipientByValidatorIndex = new MapDef<string, {epoch: Epoch; feeRecipient: string}>(() => ({
+      epoch: 0,
+      feeRecipient: opts.defaultFeeRecipient,
+    }));
+  }
+
+  add(epoch: Epoch, {validatorIndex, feeRecipient}: ProposerPreparationData): void {
```

Comment:
we should validator `epoch` here to make sure it's not too old

Reply:
`epoch` is actually coming from the clock, i guess we can skip check and avoid adding clock to this cache class. what do you think? :thinking:

Reply:
if it comes from beacon chain clock then it's fine, if it comes from api (not only validator, someone can just issue a `curl` to call the api) then we should check I think

Reply:
no api doesn't specifies this, this is added here: https://github.com/ChainSafe/lodestar/pull/3958/files#diff-b909bf67569ad6ffedfbd4ebe6c611c60e828d24b1e5d04fbafb058abc4eba50R587

Reply:
oh then it's totally fine 👍